oem/ibm/libpldmresponder/file_io_by_type.cpp

```diff
@@ -576,5 +580,50 @@ int FileHandler::readFile(const std::string& filePath, uint32_t offset,
     return PLDM_ERROR;
 }
 
+int FileHandler::readFileByFd(int fd, uint32_t offset, uint32_t& length,
+                              Response& response)
+{
+    off_t fileSize = lseek(fd, 0, SEEK_END);
+    if (fileSize == -1)
+    {
+        error("file seek failed");
+        return PLDM_ERROR;
+    }
+    if (offset >= fileSize)
+    {
+        error(
+            "FileHandler:readFileByFd:  Offset exceeds file size, OFFSET={OFFSET} FILE_SIZE={FILE_SIZE} FILE_HANDLE={FILE_HANDLE}",
+            "OFFSET", offset, "FILE_SIZE", fileSize, "FILE_HANDLE", fileHandle);
+        return PLDM_DATA_OUT_OF_RANGE;
+    }
+    if (offset + length > fileSize)
+    {
+        length = fileSize - offset;
+    }
+    auto rc = lseek(fd, offset, SEEK_SET);
+    if (rc == -1)
+    {
+        error("file seek failed");
+        return PLDM_ERROR;
+    }
+    size_t currSize = response.size();
+    response.resize(currSize + length);
+    auto filePos = reinterpret_cast<char*>(response.data());
+    filePos += currSize;
+    rc = ::read(fd, filePos, length);
+    if (rc == -1)
+    {
+        error("file read failed");
+        return PLDM_ERROR;
+    }
+    if (rc != length)
+    {
+        error(
+            "mismatch between number of characters to read and the length read, LENGTH={LEN} COUNT={COUNT}",
```

Comment:
Do we need to add any indication on what fd has this mismatch ?

Reply:
done